org.osgi.test.junit5.cm/pom.xml

```diff
@@ -0,0 +1,253 @@
+		</dependency>
+		<dependency>
+			<groupId>org.junit.platform</groupId>
+			<artifactId>junit-platform-launcher</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.junit.platform</groupId>
+			<artifactId>junit-platform-testkit</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.osgi</groupId>
+			<artifactId>osgi.core</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.osgi</groupId>
+			<artifactId>org.osgi.util.function</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.osgi</groupId>
+			<artifactId>org.osgi.util.promise</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.assertj</groupId>
+			<artifactId>assertj-core</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.osgi</groupId>
+			<artifactId>org.osgi.service.log</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.osgi</groupId>
+			<artifactId>org.osgi.service.cm</artifactId>
+		</dependency>
+		<dependency> 
+			<groupId>org.osgi</groupId>
+			<artifactId>org.osgi.test.assertj.framework</artifactId>
+		</dependency>
+		<dependency> 
+			<groupId>org.osgi</groupId>
+			<artifactId>org.osgi.test.assertj.promise</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>biz.aQute.bnd</groupId>
+			<artifactId>biz.aQute.tester.junit-platform</artifactId>
+			<version>${bnd.version}</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.felix</groupId>
+			<artifactId>org.apache.felix.logback</artifactId>
+			<version>1.0.2</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.eclipse.platform</groupId>
+			<artifactId>org.eclipse.osgi</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.felix</groupId>
+			<artifactId>org.apache.felix.configadmin</artifactId>
+			<scope>test</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>biz.aQute.bnd</groupId>
+				<artifactId>bnd-maven-plugin</artifactId>
+				<executions>
+					<!-- Integration Test Configuration -->
+					<execution>
+						<id>bnd-process-tests</id>
+						<phase>process-test-classes</phase>
+						<goals>
+							<goal>bnd-process-tests</goal>
+						</goals>
+						<configuration>
+							<artifactFragment>true</artifactFragment>
+							<bnd><![CDATA[
+							Test-Cases: org.osgi.test.junit5.cm.ConfigAnnotationTest
+							]]></bnd>
+<!-- 							Test-Cases:  -->
+<!--                            Test-Cases: org.osgi.test.junit5.cm.ConfigAnnotationTest -->
+							<includeClassesDir>false</includeClassesDir>
+							<testCases>useTestCasesHeader</testCases>
+<!--                             <testCases>${classes;HIERARCHY_INDIRECTLY_ANNOTATED;org.junit.platform.commons.annotation.Testable;CONCRETE;PUBLIC}</testCases> -->
```

Comment:
This should be `<testCases>junit5</testCases>` (which is the default anyway). You cannot specify a value other then the enum values.

Reply:
thank you. eclipse build a bar with the class...

Reply:
Heh, exactly the same issue that caught me out in bndtools/bnd#4088 If we can't easily sanity check for this in the Mojo, then perhaps we should make a loud warning about it in the documentation.

Reply:
Scratch that - I remember that I deliberately *didn't* use the default junit5 enum. There was a reason for this - the default enum includes all concrete classes in `Test-Cases`, however the JUnit 5 test bundle has lots of test case classes in that that are used as "fodder" for `EngineTestKit`. We don't want maven/Bndtools to run these fodder test cases directly. My way around it was to use a custom Test-Cases header which only included the *public* test cases and ignored the package-private ones (the default macro used by junit5 includes public and package-private classes). This allowed me to declare the fodder classes as package-private so that they wouldn't get included in `Test-Cases`. But then you have to explicitly declare the `Test-Cases` directive as part of the `<bnd>` parameter.